### ISSUE TITLE
Automated bugfix release

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#12889]
@Cadair suggested that maybe we can use https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment to fully automated bugfix releases. For instance, maybe a GitHub comment to trigger bot PR to render towncrier change log, and another comment to push to PyPI, and so on.

cc @astropy/core-release-maintainers

**Comment by Cadair** [comment]
Automate away those pesky bugfixes.

**Comment by nstarman** [comment]
Is there something conceptually wrong with linking bugfix release with each bug fix PR? Have the `bug` label do everything, kind of like the meeseeks backports.

**Comment by pllim** [comment]
A bugfix release with each PR is a little too much. While the release itself might be automated, there is still human overhead in communication and post-release activities. Also, I am not sure if it helps downstream maintainers to have to look out for a new bug fix release several times a week.

**Comment by mhvk** [comment]
Note that @astrofrog implemented part of this - at least for `pyerfa` all one has to do is push the relevant tag and the rest is automatic. Or do we have that already for astropy too?

**Comment by astrofrog** [comment]
We already have that for astropy but there are a few steps before/after that are manual.

**Comment by Cadair** [comment]
> there is still human overhead in communication and post-release activities

Boooo, automate it all!!

I think once per PR is probably too much, but getting it to the point where it's basically a human checked cron job (monthly or even weekly?!) should be the goal imo.

**Comment by pllim** [comment]
Re: https://github.com/astropy/astropy-project/issues/430#issuecomment-2605118438

Now that we use OpenAstronomy in publish workflow, what else needs to be automated?

**Comment by Cadair** [comment]
I have been hacking and testing more automated releases on my `dkist` package. The workflow I have settled on currently (there's still room for more automation) is this:

* Manually trigger the [prepare-release](https://github.com/DKISTDC/dkist/blob/main/.github/workflows/prepare-release.yml) workflow.
* A Pull Request with the rendered changelog is opened (and any other automated tasks).
* A Draft GitHub release is made, including the release notes.
* Merge the PR when ready.
* Publish the release
* The regular publish workflow runs.

**Comment by astrofrog** [comment]
> Is your specific ask

To be clear, I don't have any asks - just think it would be interesting to discuss. It would just be worth exploring whether we could make bugfix releases almost completely automatic, so we don't have to think too much about it. To go one step further than what @Cadair is suggesting, we could make it so that at periodic intervals (e.g. every week or every two weeks), a PR is automatically opened to render the changelog, and after a manual review (just to make sure everything is ok) we could merge it and this would automatically trigger a bugfix release. So this would be truly 'one-click' (unless there are issues in the changelog to fix, but these should usually be trivial).

**Comment by pllim** [comment]
If we want to go this route, what about having integration-testing to run against the bugfix RC also, just in case one of the bug fixes break downstream? It won't be the first time we thought we fixed a bug, but someone downstream was treating the bug as a feature, or when we unintentionally broke API/behavior compatibility.

**Comment by Cadair** [comment]
That sounds like a good idea, but feel orthogonal? Do we do that on every bugfix manually at the moment?

**Comment by Cadair** [comment]
I think it would be a good thing to have a different issue for 😉

**Comment by pllim** [comment]
Yes, I think you are right. When I have time, I can see if I already opened an issue about this somewhere, and if not, put some words together and open one... unless someone beats me to it.
